src/modules/profile-images/profile-images.service.ts

```diff
@@ -1,26 +1,49 @@
-import { Injectable } from '@nestjs/common';
-import { CreateProfileImageInput } from './dto/create-profile-image.input';
-import { UpdateProfileImageInput } from './dto/update-profile-image.input';
+import { forwardRef, Inject, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { resolve } from 'path';
+import { FileUpload } from 'src/common/interfaces/common.interface';
+import { Repository } from 'typeorm';
+import { Profile } from '../profiles/entities/profiles';
+import { ProfilesService } from '../profiles/profiles.service';
+import { UploadService } from '../upload/upload.service';
+import { ProfileImage } from './entities/profile-image.entity';
 
 @Injectable()
 export class ProfileImagesService {
-  create(createProfileImageInput: CreateProfileImageInput) {
-    return 'This action adds a new profileImage';
-  }
-
-  findAll() {
-    return `This action returns all profileImages`;
-  }
+  constructor(
+    @InjectRepository(ProfileImage)
+    private readonly profileImagesRepository: Repository<ProfileImage>,
+    private readonly uploadService: UploadService,
+    @Inject(forwardRef(() => ProfilesService))
+    private readonly profileService: ProfilesService
+  ) {}
 
-  findOne(id: number) {
-    return `This action returns a #${id} profileImage`;
+  async uploadProfileAvatar(
+    profileId: string, avatar: Promise<FileUpload>
+  ): Promise<ProfileImage>  {
+    const profileImage: ProfileImage = new ProfileImage();
+    profileImage.Image_Url
+    = (await this.uploadService.uploadFile(avatar, process.env.CLOUDINARY_PROFILE_FOLDER)).url;
+    profileImage.isAvatar = true;
+    
+    const profile: Profile = await this.profileService.getProfileById(profileId);
+    profileImage.Profile = profile;
```

Comment:
profile này nó là một cái id, ở entity không cần thiết thì bỏ luôn cái khỉ many to one đi